scalafix-core/shared/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -0,0 +1,97 @@
+package scalafix
+package internal.patch
+
+import scala.meta._
+
+object RangePosition {
+  def apply(position: Position): RangePosition =
+    RangePosition(position.start, position.end)
+}
+
+case class RangePosition(start: Int, end: Int) {
+  def withIn(other: RangePosition): Boolean = {
+    start <= other.start && other.end <= end
+  }
+}
+case class PointPosition(offset: Int)
+
+case class CommentFilter(
+    start: PointPosition,
+    end: Option[PointPosition],
+    rules: Set[String]
+) {
+  def isEnabled(ruleName: String, rulePosition: RangePosition): Boolean = {
+    val ruleMatches =
+      if (rules.nonEmpty) rules.contains(ruleName)
+      else true
+
+    ruleMatches && intersect(rulePosition)
+  }
+
+  private def intersect(rulePosition: RangePosition): Boolean = {
+    end match {
+      case Some(endPos) => {
+        val range = RangePosition(start.offset, endPos.offset)
+        range.withIn(rulePosition)
+      }
+      case None => {
+        start.offset <= rulePosition.start
+      }
+    }
+  }
+}
+
+sealed trait FilterAnchor {
+  def position: PointPosition
+}
+
+object FilterAnchor {
+  case class Start(position: PointPosition, rules: Set[String])
+      extends FilterAnchor
+  case class End(position: PointPosition) extends FilterAnchor
+}
+
+object EscapeHatch {
+  private val FilterAnchorStart = "scalafix:off (.*)".r
+  private val FilterAnchorEnd = "scalafix:on".r
```

Comment:
Is is possible to selectively enable a rule? `// scalafix:on Name`?

Reply:
Done.